src/model/test/HeatExchangerDesiccantBalancedFlow_GTest.cpp

```diff
@@ -0,0 +1,201 @@
+#include "../ScheduleConstant_Impl.hpp"
+#include "../Node.hpp"
+#include "../Node_Impl.hpp"
+#include "../AirLoopHVAC.hpp"
+#include "../AirLoopHVAC_Impl.hpp"
+#include "../AirLoopHVACOutdoorAirSystem.hpp"
+#include "../AirLoopHVACOutdoorAirSystem_Impl.hpp"
+#include "../HVACTemplates.hpp"
+
+using namespace openstudio;
+using namespace openstudio::model;
+
+TEST_F(ModelFixture, HeatExchangerDesiccantBalancedFlow_HeatExchangerDesiccantBalancedFlow) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  ASSERT_EXIT(
+    {
+      Model m;
+      HeatExchangerDesiccantBalancedFlow hx(m);
+      HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p(m);
+      HeatExchangerDesiccantBalancedFlow hx2(m, p);
+
+      exit(0);
+    },
+    ::testing::ExitedWithCode(0), "");
+
+  Model m;
+  HeatExchangerDesiccantBalancedFlow hx(m);
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p(m);
+  HeatExchangerDesiccantBalancedFlow hx2(m, p);
+
+  auto alwaysOn = m.alwaysOnDiscreteSchedule();
+
+  EXPECT_EQ(alwaysOn, hx.availabilitySchedule());
+  ASSERT_TRUE(hx.heatExchangerPerformance().optionalCast<HeatExchangerDesiccantBalancedFlowPerformanceDataType1>());
+  EXPECT_FALSE(hx.economizerLockout());
+
+  EXPECT_EQ(alwaysOn, hx2.availabilitySchedule());
+  ASSERT_TRUE(hx2.heatExchangerPerformance().optionalCast<HeatExchangerDesiccantBalancedFlowPerformanceDataType1>());
+  EXPECT_FALSE(hx2.economizerLockout());
+}
+
+TEST_F(ModelFixture, HeatExchangerDesiccantBalancedFlow_GettersSetters) {
+  Model m;
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p(m);
+  HeatExchangerDesiccantBalancedFlow hx(m, p);
+
+  Schedule schedule = hx.availabilitySchedule();
+  boost::optional<ScheduleConstant> scheduleConstant = schedule.optionalCast<ScheduleConstant>();
+  ASSERT_TRUE(scheduleConstant);
+  EXPECT_EQ((*scheduleConstant).value(), 1.0);
+  ASSERT_TRUE(scheduleConstant);
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 performanceObject = hx.heatExchangerPerformance();
+  EXPECT_EQ(performanceObject.name().get(), p.name().get());
+
+  ScheduleConstant scheduleConstant2(m);
+  scheduleConstant2.setValue(0.5);
+  hx.setAvailabilitySchedule(scheduleConstant2);
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p2(m);
+  hx.setHeatExchangerPerformance(p2);
+
+  Schedule schedule2 = hx.availabilitySchedule();
+  boost::optional<ScheduleConstant> scheduleConstant3 = schedule2.optionalCast<ScheduleConstant>();
+  ASSERT_TRUE(scheduleConstant3);
+  EXPECT_EQ((*scheduleConstant3).value(), 0.5);
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 performanceObject2 = hx.heatExchangerPerformance();
+  EXPECT_EQ(p2.name().get(), performanceObject2.name().get());
+}
+
+TEST_F(ModelFixture, HeatExchangerDesiccantBalancedFlow_addToNode) {
+  Model m;
+
+  AirLoopHVAC loop = addSystemType3(m).cast<AirLoopHVAC>();
+
+  AirLoopHVACOutdoorAirSystem oaSystem = loop.airLoopHVACOutdoorAirSystem().get();
```

Comment:
Interesting. Don't really care, but it's not much shorter than https://github.com/NREL/OpenStudio/blob/1799f9c303307c0e7adae3eda006054eebc376b9/src/model/test/ZoneHVACTerminalUnitVariableRefrigerantFlow_GTest.cpp#L211-L216